README.md

```diff
@@ -180,3 +180,74 @@ Parameters:
 * `debug` - optionally allows to start the JVM that is used to generated with a debugger. Setting it to `true` will cause
   the started JVM to suspend until a debugger is attached. Useful for debugging classloading problems or exceptions during
   the build.
+
+## Model Check
+
+Run the model check on a subset or all models in a project directly from gradle.
+
+This functionality currently runs all model checks (typesystem, structure, constrains, etc.) from gralde. By default if
+any of checks fails the complete build is failed. All messages (Info, Warning or Error) are reported through log4j to
+the command line.
+
+### Usage
+
+A minimal build script to check all models in a MPS project with no external plugins would look like this: 
+
+```
+apply plugin: 'modelcheck'
+
+configurations {
+    mps
+}
+
+dependencies {
+    mps "com.jetbrains:mps:$mpsVersion"
+}
+
+ext.mpsVersion = '2018.2.5'
+
+modelcheck {
+    projectLocation = new File("./mps-prj")
+    mpsConfig = configurations.mps
+}
+```
+
+Parameters:
+* `mpsConfig` - the configuration used to resolve MPS. Currently only vanilla MPS is supported and no custom RCPs.
+  Custom plugins are supported via the `pluginLocation` parameter.
+* `mpsLocation` - optional location where to place the MPS files.
+* `plugins` - optional list of plugins to load before generation is attempted.
+  The notation is `new Plugin("someID", "somePath")`. Where the first parameter is the plugin id and the second the `short (folder) name`.
+* `pluginLocation` - location where to load the plugins from. Structure needs to be a flat folder structure similar to the
+  `plugins` directory inside of the MPS installation.
+* `models` - optional list of models to generate. If omitted all models in the project will be generated. Only full name
+  matched are supported and no RegEx or partial name matching.
+* `macros` - optional list of path macros. The notation is `new Macro("name", "value")`.
+* `projectLocation` - location of the MPS project to generate.
+* `errorNoFail` - report errors but do not fail the build.
+* `warningsAsError` - handles warnings as errors and will fail the build if any is found when `errorNoFail` is not set. 
```

Comment:
There's an inconsistency between `--warning-as-error` and `warningsAsError` setting (the latter has an 's').